### PR TITLE
Support remove flag for kubectl run command

### LINKED ISSUE #16672: Kubectl --rm flag (like docker)
I find that I frequently want to run a command in my cluster to test something.  When that command is done, I want the pod to be removed.  Soemthing like:

```
$ kubectl run date -i --tty --rm --restart=Never --image=busybox -- nslookup foobar
```

### PR DESCRIPTION
**Pull request by feihujiang**
Closes #16672

### CODE REVIEW

File: pkg/kubectl/cmd/run.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func addRunFlags(cmd *cobra.Command) {
 	cmd.Flags().String("image", "", "The image for the container to run.")
 	cmd.MarkFlagRequired("image")
 	cmd.Flags().IntP("replicas", "r", 1, "Number of replicas to create for this container. Default is 1.")
+	cmd.Flags().Bool("rm", false, "If true, delete the object created in this command when interactive mode ended.")
```
Comment (deads2k) [inline pkg/kubectl/cmd/run.go]: Add validation that prevents using `--rm` when you're not attached to match `docker run`

File: pkg/kubectl/cmd/run.go [hunk 2]
```diff
@@ -213,7 +214,32 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
...
 		}
-		return handleAttachPod(f, client, attachablePod, opts)
+		err = handleAttachPod(f, client, attachablePod, opts)
+		if err != nil {
+			return err
+		}
+
+		remove := cmdutil.GetFlagBool(cmd, "rm")
+		if remove {
+			reaper, err := f.Reaper(mapping)
```
Comment (deads2k) [inline pkg/kubectl/cmd/run.go]: Can you call `ReapResult` from here to avoid duplicating deletion logic?

**Comment by deads2k** [comment]
This looks like it would be hard to unit test, but can you add a test for this capability in `test-cmd.sh`.

File: pkg/kubectl/cmd/run.go (hunk 1 above)
Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: docker run --rm is effective whatever the mode is, we should discuss whether and how to support that firstly in the issue #16672

**Comment by feihujiang** [comment]
@deads2k , thanks for quick review. OK, I'll add a test.

File: pkg/kubectl/cmd/run.go (hunk 2 above)
Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: OK, I'll fix this.

**Comment by feihujiang** [comment]
When we make an agreement which level we should support --rm flag after discussion in the issue #16672. I'll address the comments and update this PR.

File: pkg/kubectl/cmd/run.go (hunk 1 above)
Comment (deads2k) [inline pkg/kubectl/cmd/run.go]: > docker run --rm is effective whatever the mode is, we should discuss whether and how to support that firstly in the issue #16672

Based on the doc, https://docs.docker.com/engine/reference/run/#detached-d, you can't use `--rm` in combination with `-d` (detached).

**Comment by feihujiang** [comment]
~~~
!!! Error in ./hack/test-update-storage-objects.sh:48
  'return 1' exited with status 1
Call stack:
  1: ./hack/test-update-storage-objects.sh:48 startApiServer(...)
  2: ./hack/test-update-storage-objects.sh:122 main(...)
Exiting with status 1
~~~
Jenkins unit/integration failure is not refer to this PR.
@k8s-bot test this please

File: pkg/kubectl/cmd/run.go (hunk 1 above)
Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: done

File: pkg/kubectl/cmd/run.go (hunk 2 above)
Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: Done

**Comment by feihujiang** [comment]
@deads2k  PTAL. Comments are addressed, but ```test-cmd.sh``` is just for simple scenarios, it isn't for this capability, so I add an e2e test for this  capability.

**Comment by deads2k** [comment]
lgtm

**Pull request merged**